.github/workflows/ci.yml

```diff
@@ -0,0 +1,138 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+    tags:
+      - "*"
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  ci:
+#    name: ${{ env.JOB_NAME }}
+    name: ${{ matrix.task.name}} - ${{ matrix.python.name }}
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        python:
+          - name: CPython 2.7
+            tox: py27
+            action: 2.7
+          - name: CPython 3.5
+            tox: py35
+            action: 3.5
+          - name: CPython 3.6
+            tox: py36
+            action: 3.6
+          - name: CPython 3.7
+            tox: py37
+            action: 3.7
+        task:
+          - name: Test
+            tox: tests
+            type: test
```

Comment:
I did rework some aspects to be more similar to your suggestion but this did in part stick around.  It serves little purpose specifically in the test job but it does retain symmetry with the added check job where it is used.  I like to retain similarity in job matrix layouts so that there's less to think about when copying bits from one to another in the same workflow.